src/features/rockets/rocketsSlice.js

```diff
@@ -0,0 +1,50 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const url = 'https://api.spacexdata.com/v4/rockets';
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', async (_, { rejectWithValue }) => {
+  try {
+    const response = await axios.get(url);
+    return response.data;
+  } catch (error) {
+    return rejectWithValue('Failed to fetch rockets data !!');
+  }
+});
+
+const initialState = {
+  rocketItems: [],
+  isLoading: true,
+  error: undefined,
+};
+
+const rocketsSlice = createSlice({
+  name: 'rockets',
+  initialState,
+  reducers: {},
+  extraReducers(builder) {
+    builder
+      .addCase(fetchRockets.pending, (state) => {
```

Comment:
Very good separation of promises. My recommendation would be to reset the error value in every case a change in status is made. Remember, the first time there might be an issue fetching the data, if you try a second time, your slice is going to remember the previous error value store in your slice

Reply:
Awesome, thank you. The required change has been made.